src/sugar/extensions/swarm.py

```diff
@@ -0,0 +1,532 @@
+            'logs',
+            services=services_names,
+            options_args=options_args,
+        )
+
+    @docparams(doc_common_no_services)
+    def _cmd_ls(
+        self,
+        options: str = '',
+    ) -> None:
+        """List services."""
+        options_args = self._get_list_args(options)
+        self._call_service_command('ls', options_args=options_args)
+
+    @docparams(doc_common_services)
+    def _cmd_ps(
+        self,
+        services: str = '',
+        all: bool = False,
+        options: str = '',
+    ) -> None:
+        """List the tasks of one or more services."""
+        services_names = self._get_services_names(services=services, all=all)
+        options_args = self._get_list_args(options)
+        self._call_service_command(
+            'ps', services=services_names, options_args=options_args
+        )
+
+    @docparams(doc_common_services)
+    def _cmd_rm(
+        self,
+        services: str = '',
+        all: bool = False,
+        options: str = '',
+    ) -> None:
+        """Remove one or more services."""
+        services_names = self._get_services_names(services=services, all=all)
+        options_args = self._get_list_args(options)
+        self._call_service_command(
+            'rm', services=services_names, options_args=options_args
+        )
+
+    @docparams(doc_common_services)
+    def _cmd_rollback(
+        self,
+        services: str = '',
+        all: bool = False,
+        options: str = '',
+    ) -> None:
+        """Revert changes to a service's configuration."""
+        services_names = self._get_services_names(services=services, all=all)
+        options_args = self._get_list_args(options)
+        self._call_service_command(
+            'rollback',
+            services=services_names,
+            options_args=options_args,
+        )
+
+    @docparams(doc_common_services)
+    def _cmd_scale(
+        self,
+        services: str = '',
+        all: bool = False,
+        options: str = '',
+    ) -> None:
+        """Scale one or multiple replicated services."""
+        services_names = self._get_services_names(services=services, all=all)
+        options_args = self._get_list_args(options)
+        self._call_service_command(
+            'scale',
+            services=services_names,
+            options_args=options_args,
+        )
+
+    @docparams(doc_common_services)
+    def _cmd_update(
+        self,
+        services: str = '',
+        all: bool = False,
+        options: str = '',
+    ) -> None:
+        """Update a service."""
+        services_names = self._get_services_names(services=services, all=all)
+        options_args = self._get_list_args(options)
+        self._call_service_command(
+            'update',
+            services=services_names,
+            options_args=options_args,
+        )
+
+    # Node commands
+    @docparams(doc_node_options)
+    def _cmd_node(
+        self,
+        demote: str = '',
```

Comment:
now, the positional only is already implemented .. 
so it should look like this:

```python
    def _cmd_node(
        self,
        action: str = '',
        /,
        update: str = '',
        options: str = '',
    ) - > None:
```

Reply:
remember to update **doc_node_options**

Reply:
I think we still need to address this. but we can do it in follow up prs